src/AbstractProxyFixer.php

```diff
@@ -23,47 +24,71 @@
 abstract class AbstractProxyFixer extends AbstractFixer
 {
     /**
-     * @var FixerInterface
+     * @var FixerInterface[]
      */
-    protected $proxyFixer;
+    protected $proxyFixers;
 
     public function __construct()
     {
-        parent::__construct();
+        foreach (Utils::sortFixers($this->createProxyFixers()) as $proxyFixer) {
+            $this->proxyFixers[$proxyFixer->getName()] = $proxyFixer;
+        }
 
-        $this->proxyFixer = $this->createProxyFixer();
+        parent::__construct();
     }
 
     /**
      * {@inheritdoc}
      */
     public function isCandidate(Tokens $tokens)
     {
-        return $this->proxyFixer->isCandidate($tokens);
+        foreach ($this->proxyFixers as $fixer) {
+            if ($fixer->isCandidate($tokens)) {
+                return true;
+            }
+        }
+
+        return false;
     }
 
     /**
      * {@inheritdoc}
      */
     public function isRisky()
     {
-        return $this->proxyFixer->isRisky();
+        foreach ($this->proxyFixers as $fixer) {
+            if ($fixer->isRisky()) {
+                return true;
+            }
+        }
+
+        return false;
     }
 
     /**
      * {@inheritdoc}
      */
     public function getPriority()
     {
-        return $this->proxyFixer->getPriority();
+        if (count($this->proxyFixers) > 1) {
+            throw new \LogicException('You need to override this method to provide the priority of combined fixers.');
+        }
+
+        return reset($this->proxyFixers)->getPriority();
     }
 
     /**
      * {@inheritdoc}
      */
     public function supports(\SplFileInfo $file)
     {
-        return $this->proxyFixer->supports($file);
+        foreach ($this->proxyFixers as $fixer) {
+            if ($fixer->supports($file)) {
+                return true;
```

Comment:
all proxied fixers need to support file, here, you implement "at least one fixer" rule

Reply:
no, let us do it other way. i got your idea here